### ISSUE TITLE
The spring.cloud.stream.default configuration is not functioning as expected

### ISSUE THREAD

**Issue by jbharper86** [spring-cloud/spring-cloud-stream#1522]
The `spring.cloud.stream.default` configuration is no longer treated as the default values for individual streams as of spring-cloud-stream version `Elmhurst.SR1`/`2.0.1.RELEASE`. I have confirmed that this was working in `Elmhurst.RELEASE`/`2.0.0.RELEASE`.

According to the [docs](https://docs.spring.io/spring-cloud-stream/docs/Elmhurst.SR1/reference/html/_configuration_options.html#binding-properties), common configuration options can be placed in this location to act as defaults for streams. However, these options are ignored in the latest release.

I believe the issue is in `EnvironmentEntryInitializingTreeMap`. In the latest release, `BindingProperties` was updated to implement the `MergableProperties` interface. This caused a change in the `EnvironmentEntryInitializingTreeMap.put` method:

```
// Before
@Override
public T put(String key, T value) {
	// boot 2 call this first
	Binder binder = new Binder(ConfigurationPropertySources.get(environment),new PropertySourcesPlaceholdersResolver(environment),this.conversionService, null);
	binder.bind(defaultsPrefix, Bindable.ofInstance(value));
	return this.delegate.put(key, value);
}
```	
```
// After
@Override
public T put(String key, T value) {
	Binder binder = new Binder(ConfigurationPropertySources.get(environment),new PropertySourcesPlaceholdersResolver(environment),this.conversionService, null);
	T defaultProperties = BeanUtils.instantiateClass(entryClass);
	binder.bind(defaultsPrefix, Bindable.ofInstance(defaultProperties));
	((MergableProperties)defaultProperties).merge((MergableProperties) value);
	return this.delegate.put(key, value);
}
```

The merge line in the new method appears to be reversed. Instead of merging from the defaultProperties to the value object, it seems to be copying from the value to the defaultProperties.

**Comment by olegz** [comment]
Actually this has been fixed and validated (see https://github.com/spring-cloud/spring-cloud-stream/issues/1507), but you need to upgrade to just release 2.1.0.RC1.

**Comment by jbharper86** [comment]
Any chance that fix can be backported to the 2.0.x line? Upgrading to 2.1.0 isn't really an option for us at the moment.

**Comment by olegz** [comment]
@jbharper86 unfortunately this change propagates all the way to Spring Boot. In other words Spring Boot teams exposed certain hooks for us and those hooks are only available in boot 2.1.0. So back porting this would not be possible.
